<img src="http://www.neomades.com/extern/partage/ecoCode/2sur5_1x.png">
<p>Turning on the torch mode programmatically must absolutely be avoided because the flashlight is one of the most energy-intensive component.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    let device = AVCaptureDevice.default(for: AVMediaType.video)
    device.torchMode = .on
</pre>
<pre>
    let device = AVCaptureDevice.default(for: AVMediaType.video)
    device.setTorchModeOn(level: AVCaptureDevice.maxAvailableTorchLevel / 2.0)
</pre>
<pre>
    let device = AVCaptureDevice.default(for: AVMediaType.video)
    device.torchMode = AVCaptureTorchMode.on
</pre>